indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamEndSequenceNumbers.java

```diff
@@ -147,6 +148,39 @@ public SeekableStreamSequenceNumbers<PartitionIdType, SequenceOffsetType> plus(
     }
   }
 
+  @Override
+  public boolean isGreater(
+          SeekableStreamSequenceNumbers<PartitionIdType, SequenceOffsetType> other
+  )
+  {
+    if (this.getClass() != other.getClass()) {
+      throw new IAE(
+        "Expected instance of %s, got %s",
+        this.getClass().getName(),
+        other.getClass().getName()
+      );
+    }
+
+    final SeekableStreamEndSequenceNumbers<PartitionIdType, SequenceOffsetType> otherStart =
+            (SeekableStreamEndSequenceNumbers<PartitionIdType, SequenceOffsetType>) other;
+
+    if (stream.equals(otherStart.stream)) {
+      //Same stream, compare the offset
+      AtomicReference<Boolean> res = new AtomicReference<>(false);
+      partitionSequenceNumberMap.forEach(
+          (partitionId, sequenceOffset) -> {
+            if (otherStart.partitionSequenceNumberMap.get(partitionId) != null && Long.parseLong(String.valueOf(sequenceOffset)) > Long.parseLong(String.valueOf(otherStart.partitionSequenceNumberMap.get(partitionId)))) {
```

Comment:
this is making a lot of assumptions about sequence offsets and whether they can be parsed as long values. While this is true for Kafka, this is not necessarily true for other supervisor implementations (e.g. Kinesis) so we will probably need to generalize this

Reply:
@xvrl, When i was directly converting int to long it was failing for kinesis test cases, But after converting to string and then to Long, All kinesis and Kafka test cases are passing.

Reply:
that's not a very scientific way to verify this. I think we should have clear type constraints that allow us to do what we do here and not just hope that it works for all cases. We may not be covering all cases in tests. From reading the code it looks like Kinesis has string offsets which are not numbers.